Apps/Sandcastle/gallery/Manually Controlled Animation.html

```diff
@@ -0,0 +1,167 @@
+            start,
+            factor * totalSeconds,
+            new Cesium.JulianDate()
+          );
+
+          // Lerp using a non-linear factor so that the model accelerates.
+          const locationFactor = Math.pow(factor, 2);
+          const location = Cesium.Cartesian3.lerp(
+            startPosition,
+            endPosition,
+            locationFactor,
+            new Cesium.Cartesian3()
+          );
+          position.addSample(time, location);
+          distance.addSample(
+            time,
+            (totalDistance += Cesium.Cartesian3.distance(
+              location,
+              prevLocation
+            ))
+          );
+          prevLocation = location;
+        }
+
+        function updateSpeedLabel(time, result) {
+          velocityVectorProperty.getValue(time, velocityVector);
+          const metersPerSecond = Cesium.Cartesian3.magnitude(velocityVector);
+          const kmPerHour = Math.round(metersPerSecond * 3.6);
+
+          return `${kmPerHour} km/hr`;
+        }
+
+        // Add our model.
+        const modelPrimitive = viewer.scene.primitives.add(
+          Cesium.Model.fromGltf({
+            url: "../../SampleData/models/CesiumMan/Cesium_Man.glb",
+            scale: 4,
+          })
+        );
+        const modelLabel = viewer.entities.add({
+          position: position,
+          orientation: new Cesium.VelocityOrientationProperty(position), // Automatically set the model's orientation to the direction it's facing.
+          label: {
+            text: new Cesium.CallbackProperty(updateSpeedLabel, false),
+            font: "20px sans-serif",
+            showBackground: true,
+            distanceDisplayCondition: new Cesium.DistanceDisplayCondition(
+              0.0,
+              100.0
+            ),
+            eyeOffset: new Cesium.Cartesian3(0, 7.2, 0),
+          },
+        });
+
+        modelPrimitive.readyPromise.then(function (model) {
+          model.activeAnimations.addAll({
+            loop: Cesium.ModelAnimationLoop.REPEAT,
+            animationTime: function (duration) {
+              return distance.getValue(viewer.clock.currentTime) / duration;
+            },
+            multiplier: 0.25,
+          });
+          const rot = new Cesium.Matrix3();
+          viewer.scene.preUpdate.addEventListener(function () {
+            const time = viewer.clock.currentTime;
+            const pos = position.getValue(time);
+            const vel = velocityVectorProperty.getValue(time);
+            Cesium.Cartesian3.normalize(vel, vel);
+            Cesium.Transforms.rotationMatrixFromPositionVelocity(
+              pos,
+              vel,
+              viewer.scene.globe.ellipsoid,
+              rot
+            );
+            Cesium.Matrix4.fromRotationTranslation(rot, pos, model.modelMatrix);
+          });
+        });
+        viewer.trackedEntity = modelLabel;
+        modelLabel.viewFrom = new Cesium.Cartesian3(-20.0, -7.0, 4.0);
```

Comment:
```suggestion
        modelLabel.viewFrom = new Cesium.Cartesian3(-30.0, -10.0, 10.0);
```
Small tweak, these `Cartesian3` values give better visibility of the model when it first loads in.